Kernel/Storage/AHCIPort.cpp

```diff
@@ -609,25 +651,34 @@ bool AHCIPort::identify_device()
     if (!spin_until_ready())
         return false;
 
-    full_memory_barrier();
-    mark_command_header_ready_to_process(unused_command_header.value());
-    full_memory_barrier();
-
-    while (1) {
-        if (m_port_registers.serr != 0) {
-            dbgln("AHCI Port {}: Identify failed, SError 0x{:08x}", representative_port_index(), (u32)m_port_registers.serr);
-            try_disambiguate_sata_error();
-            return false;
+    // FIXME: Find a better way to send IDENTIFY DEVICE and getting an interrupt!
+    {
+        main_lock.unlock();
```

Comment:
Since `main_lock.unlock()` is supposed to re-enable interrupts, we should probably `VERIFY_INTERRUPTS_ENABLED` here to make sure this actually happened.